From 1ff24046864714db3e4e8c9719132ac835732918 Mon Sep 17 00:00:00 2001
From: Michael Ortner <lineage@michael-ortner.de>
Date: Tue, 31 Dec 2019 17:29:12 +0100
Subject: [PATCH] Temporary fix sepolicy build issues

due to sysfs_disk_stat and wpa_data_file
objects not declared error.

Change-Id: Ieb7f0de017d6294f30c5cfd8fef342395326da35
---
 private/compat/28.0/28.0.cil        | 1 +
 private/compat/28.0/28.0.ignore.cil | 4 +++-
 vendor/file.te                      | 1 +
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/private/compat/28.0/28.0.cil b/private/compat/28.0/28.0.cil
index 305cb3ac..7f8f6a0d 100644
--- a/private/compat/28.0/28.0.cil
+++ b/private/compat/28.0/28.0.cil
@@ -18,6 +18,7 @@
 (type thermalserviced_exec)
 (type untrusted_v2_app)
 (type vcs_device)
+(type wpa_data_file)
 
 ;; Public 28.0 SEPolicy is divergent on different devices w.r.t
 ;; exported_audio_prop type. We need this typeattribute declaration so that the
diff --git a/private/compat/28.0/28.0.ignore.cil b/private/compat/28.0/28.0.ignore.cil
index 98c4b9c9..d2dae667 100644
--- a/private/compat/28.0/28.0.ignore.cil
+++ b/private/compat/28.0/28.0.ignore.cil
@@ -121,6 +121,7 @@
     simpleperf_app_runner_exec
     su_tmpfs
     super_block_device
+    sysfs_disk_stat
     sysfs_fs_f2fs
     system_bootstrap_lib_file
     system_event_log_tags_file
@@ -147,4 +148,5 @@
     vendor_misc_writer_exec
     vendor_task_profiles_file
     vrflinger_vsync_service
-    watchdogd_tmpfs))
+    watchdogd_tmpfs
+    wpa_data_file))
diff --git a/vendor/file.te b/vendor/file.te
index 0b1fd748..d3527e5c 100644
--- a/vendor/file.te
+++ b/vendor/file.te
@@ -1,2 +1,3 @@
 type hostapd_data_file, file_type, data_file_type;
 type wpa_data_file, file_type, data_file_type;
+type sysfs_disk_stat, fs_type, sysfs_type;
-- 
2.17.1

